### DISCUSSION TITLE
Filter Search Results based on SearchTerms

### DISCUSSION THREAD

**Discussion by Srijit1998** [microsoft-search/pnp-modern-search#3534] [Q&A]
I am using PnP Modern Search v4.5. Currently, whenever any user is entering '230' as searchTerm, I am getting results that have backend IDs as '203' in my Search Results.

So, to eliminate those results, I have written a query in Query Template as below:

{searchTerms} ListItemID<>{searchTerms}

This is working fine if the user is only entering the searchTerm as '230'. But, if the user is entering 'id 230' as searchTerm, then I am getting the results that have backend IDs as '230'.

Is there any way to check whether the listItemID is present in {searchTerms} so that I can filter those items and get the desired results.
, i.e. I don't want results that have the listItemID as present in search term entered by the user.

I am new to pnp search, is there anything that I can write in Query Template or anywhere that meets my requirement?

Can anyone please help me on this? Any help would be appreciated. Thanks!

@wobba @kasperbolarsen @patrikhellgren

**Comment by kasperbolarsen** [comment]
I think that it would be easier to assist if you could define what you are searching for rather than what you don't want to see.
Are you searching for 'everything but listitems where the ID matches the search query '

**Reply by Srijit1998** [reply]
Thanks @kasperbolarsen for the reply. Yes, I am searching for everything, except the list items where the ID is present in the search query.
For eg. search query - 'id 245' , I need everything as results except those items that have the backend IDs as '245'.

**Reply by kasperbolarsen** [reply]
In that case your existing query is as good as it gets, unless you will be able to pinpoint the properties to search, like 
Title:{searchTerms} OR People:{searchTerms}    and leaving OriginalPath out of the query.

**Reply by Srijit1998** [reply]
Sorry, I didn't get you.
Here is my search query.

<img width="596" alt="image" src="https://github.com/microsoft-search/pnp-modern-search/assets/71933801/1ccb6f6b-4dd6-47f3-a807-c367852db0d7">

Here is my query template

<img width="215" alt="image" src="https://github.com/microsoft-search/pnp-modern-search/assets/71933801/74b7cb9e-02af-4f7e-b6b5-54147e844377">

As per the above query template rule, I am getting 6 results that have backend IDs as '203'. I don't want these 6 items. Is there any way to check if ListItemID is present in searchTerms so that I can eliminate the results having the ListItemIDs if that is present in searchTerm.

@kasperbolarsen

**Comment by wobba** [comment]
Seems weird that 230 returns result on 203. Can you isolate a repro issue for this? Did a random test myself without being able to trigger this.

**Reply by Srijit1998** [reply]
My search results are working fine if I am using the search query as '203', i.e a single word.

ListItemID<>{{searchTerms}} as 203 != 203 is false , so I am not getting the items with IDs as 203 in backend lists.

But, when I am searching with multiple words, for eg 'id 203', I am getting all the items along with the items that have backend IDs as 203 in backend lists.

ListItemID<>{{searchTerms}} , as 203 != 'id 203' is true, I am getting the items with IDs 203 as well.

I want to ensure that searchTerms or search query does not contain the ListItemID property value so that I can eliminate those items in the results.

@kasperbolarsen @wobba

**Reply by wobba** [reply]
I think I know what happens as the `Path`property is now searchable, and it contains `DispForm.aspx?id=203` matching on those terms. And there is no good solution as you have no guarantee that a list item should not be returned as it may be tagged with that term.

I have reached out to the backend team about this case of URL's to see if there is something we can do.

In your orginal issue you said `230` gave recall on `203`. Is that the issue or misspelling?

**Reply by Srijit1998** [reply]
Yes, absolutely correct. The 'OriginalPath' property contains DispForm.aspx?ID=203 and it matches with my query. Is there any way to dynamically check whether 'OriginalPath' property contains the ID that is provided in the searchTerm?

Thanks @wobba , please let me know if there is any response from the backend team on this.

**Reply by wobba** [reply]
@Srijit1998 you can exclude if the property has a value, but you have no guarantee that you would like the result back as the term could match in some text for the item. What kind of items are you returning in search? Depending on your scenario you may can work around it.

**Reply by Srijit1998** [reply]
@wobba If '203' appears in any item content and that item appears in one of the search results, then that is fine for me. But, for me, I am getting some of the items as results where '203' is not mentioned anywhere in item content or anywhere else but, its present as ID column value in one of our sharepoint lists and I don't want anyone to get the results where the search term is not present anywhere in the content but its present in ID column value of sharepoint lists.

I could eliminate this issue with single word search, but, with multiple words search, I am not able to do it.

The items that I am returning in search are local sharepoint results.

**Reply by wobba** [reply]
"Local sharepoint results" is very generic. Sites, pages, lists, list items, files? All of them?
